### ISSUE TITLE
AttributeError: 'Retry' object has no attribute 'method_whitelist'

### ISSUE THREAD

**Issue by ratanlp** [jugaad-py/jugaad-trader#11]
When trying to fetch any of the data like kite.margins() or kite.orders(), I'm getting error
AttributeError: 'Retry' object has no attribute 'method_whitelist'

Traceback (most recent call last):
  File "/Users/usr/Documents/Projects/Python//jugaad_trader_sample.py", line 58, in <module>
    margins = kite.margins()
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/kiteconnect/connect.py", line 315, in margins
    return self._get("user.margins")
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/kiteconnect/connect.py", line 817, in _get
    return self._request(route, "GET", params)
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/jugaad_trader/zerodha.py", line 167, in _request
    raise e
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/jugaad_trader/zerodha.py", line 163, in _request
    proxies=self.proxies)
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/requests/sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/urllib3/connectionpool.py", line 805, in urlopen
    if retries.is_retry(method, response.status, has_retry_after):
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/urllib3/util/retry.py", line 343, in is_retry
    if not self._is_method_retryable(method):
  File "/Users/usr/opt/anaconda3/lib/python3.7/site-packages/urllib3/util/retry.py", line 331, in _is_method_retryable
    if self.method_whitelist and method.upper() not in self.method_whitelist:
AttributeError: 'Retry' object has no attribute 'method_whitelist'

How do I solve this?

**Comment by ratanlp** [comment]
Nevermind. Solved after adding following in requirements.txt.
requests==2.23.0
urllib3==1.25.11

**Comment by sevakram** [comment]
How do I replicat3 the issue? this needs to be fixed permanently if its a general issue

requests==2.23.0 should be installed as dependency..

Please specify-
Operating system:
Python version:
Are you using virtual environment?

**Comment by ratanlp** [comment]
I'm on MacOS with Python version - 3.7.9.
Earlier I had older version of jugaad trader and it was working fine.
Few days back I updated jugaadTrader to new version and after that I started getting the above error.

request==2.23.0 requires urllib3 to be on or below 1.25.11 and my version for urllib3==1.26.3 was causing the issue.

After updating the requirements.txt file, it started working.

Almost forgot to thank you for this awesome project. Thanks a lot :)
